packages/webassembly-playground/src/lsp/index.ts

```diff
@@ -0,0 +1,285 @@
+import type * as monaco from "monaco-editor";
+import {
+  MonacoToProtocolConverter,
+  ProtocolToMonacoConverter,
+} from "@codingame/monaco-languageclient";
+import init, { LanguageServer } from "../../../../lsp/pkg/lsp";
+
+export type LanguageServerWrapper = monaco.IDisposable & {
+  commit(uri: string, content: string): void;
+  pullDiagnostics(model: monaco.editor.ITextModel): any;
+};
+
+export async function startLanguageServer(): Promise<LanguageServerWrapper> {
+  const [monaco] = await Promise.all([import("monaco-editor"), init()]);
+  const languageServer = new LanguageServer();
+
+  monaco.languages.register({ id: "wat", extensions: [".wat"] });
+  monaco.languages.setMonarchTokensProvider("wat", {
+    brackets: [{ open: "(", close: ")", token: "delimiter.parenthesis" }],
+    keywords: [
+      "module",
+      "func",
+      "type",
+      "param",
+      "result",
+      "local",
+      "end",
+      "if",
+      "then",
+      "else",
+      "block",
+      "loop",
+      "data",
+      "elem",
+      "declare",
+      "export",
+      "global",
+      "memory",
+      "import",
+      "start",
+      "table",
+      "item",
+      "offset",
+      "mut",
+    ],
+    typeKeywords: ["i32", "i64", "f32", "f64", "v128", "funcref", "externref"],
```

Comment:
Does the wat-language-server support `funcref` and `externref` now?

Reply:
Just behaves like number types and vector type, nothing else.